dwave/cloud/config/loaders.py

```diff
@@ -116,6 +117,42 @@ def parse_boolean(s, default=None):
     return bool(s)
 
 
+def _solver_id_as_identity(id: str) -> dict:
+    """Construct a solver identity dictionary from the unique string representation
+    generated with :func:`_dump_solver_id`.
+
+    Note: for internal use only. For public interface, see
+    :meth:`dwave.cloud.api.models.SolverIdentity.from_id`.
+    """
+    name, *version = id.strip().split(';')
+    identity = dict(name=unquote(name))
+    if not identity['name']:
+        raise ValueError('Invalid id string: missing "name"')
+
+    version = dict(map(unquote, v.split('=', maxsplit=1)) for v in version)
+    identity.update(**dict(version=version) if version else {})
```

Comment:
Why not just
```suggestion
    identity.update({"version": version} if version else {})
```
or even (although it's perhaps a bit convoluted)
```suggestion
    if version := dict(map(unquote, v.split('=', maxsplit=1)) for v in version):
        identity["version"] = version
```